src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -194,23 +193,26 @@ private Cursor readCursor(JsonParser jsonParser) throws IOException {
         final Iterator<T> eventIterator = eventReader.readValues(jsonParser, eventClass);
 
         final List<T> events = new ArrayList<>();
-        while (eventIterator.hasNext()) {
-            readEvent(eventIterator, events);
-        }
-        return events;
-    }
-
-    private void readEvent(final Iterator<T> source, final List<T> target) throws JsonMappingException {
-        try {
-            target.add(source.next());
-        } catch (final RuntimeJsonMappingException e) {
-            final Throwable cause = e.getCause();
-            if (cause instanceof JsonMappingException) {
-                listener.onMappingException((JsonMappingException) cause);
-            } else {
-                throw e;
+        while (true) {
+            try {
+                // MappingIterator#hasNext can theoretically also throw RuntimeExceptions, that's why we use this strange loop structure
```

Comment:
See https://github.com/FasterXML/jackson-databind/blob/jackson-databind-2.7.6/src/main/java/com/fasterxml/jackson/databind/MappingIterator.java#L180 and https://github.com/FasterXML/jackson-databind/blob/jackson-databind-2.7.6/src/main/java/com/fasterxml/jackson/databind/MappingIterator.java#L409

Not sure in what cases it can really happen, but according to the code it's possible.